src/groupByKey.ts

```diff
@@ -1,7 +1,22 @@
-type GroupsMap<T> = {
-  [key: string]: T[];
+type GroupsMap<Group> = {
+  [key: string]: Group[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<Group>(
+  items: Group[],
+  key: keyof Group,
+): GroupsMap<Group> {
+  const grouppedItems: GroupsMap<Group> = items.reduce((result, current) => {
+    const itemKey = current[key] as string;
```

Comment:
Casting `current[key]` to a string assumes that the key will always be a string property, which might not be the case for all possible uses of this function. It would be safer to ensure that the function can only be called with keys that are guaranteed to be strings, or handle non-string keys appropriately.